mockgcp: Implement compute addresses

justinsb opened the pull request.

justinsb commented:
/assign @maqiuyujoyce 

(I noticed that we don't have test coverage for the address-in-status change)

maqiuyujoyce reviewed (commented):
LGTM! Thank you for supporting the Compute address mock. A couple of questions for self-education purpose:

1. Just to confirm: We don't need to do the merge for locational resources in the mock because we redirect calls made by TF to the GCP APIs from the real backend to the mock, right?
2. You mentioned that this mock is to help test the newly added `status.observedState.address` field. Though I think we don't have a good verification step for status values. Should I consider this PR a prerequisite for such verification then?

justinsb commented on mockgcp/mockcompute/service.go:
```diff
@@ -37,9 +37,6 @@ type MockService struct {
 	operations *operations.Operations
-
-	networksv1 *NetworksV1
-	subnetsv1  *SubnetsV1
```
So they aren't resources, they are proto services.  I'm trying your suggestion of splitting the files by resource instead (even when there are multiple resources in a proto service), that seems to work well.

justinsb commented:
> I think we want to add an export test here. I might add it now...

Just cleaned up and rebased.  We need more PRs to land (workflows) before we can test this one consistently with GCP

maqiuyujoyce commented:
/lgtm though the mock test need to pass first.

It looks like the mock test is sometimes flaky. @justinsb do you think it is something we'll look into?

justinsb commented:
>It looks like the mock test is sometimes flaky. @justinsb do you think it is something we'll look into?

Yes.  @gemmahou is fixing the deletion ordering non-determinism, I'm fixing the sample-loading non-determinism

justinsb commented:
This one is blocking #1260, which blocks alloydb mocking.  Mind taking another look @maqiuyujoyce ?

maqiuyujoyce commented:
/lgtm

maqiuyujoyce commented:
/approve
